storage: misc logging improvements

tbg opened the pull request:
Got confused by this message in #28995. Prior to this
commit on a merge it looks like the subsuming range is
removing itself, but it's actually the subsumee.

Release note: None

tbg committed:
storage: fix replica removal log buglet

Got confused by this message in #28995. Prior to this
commit on a merge it looks like the subsuming range is
removing itself, but it's actually the subsumee.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/29000)
<!-- Reviewable:end -->

tbg committed:
storage: denoise RelocateRange

This very spammy log message got in the way of the investigation in #28995.

Release note: None

tbg committed:
storage: log when snapshot is used, not created

This was another source of noise in #28995. It makes more sense to log a
snapshot in either the error when it fails or when we actually start
streaming it.

Release note: None

tbg committed:
storage: remove overly severe log message

This doesn't seem worth logging (definitely not at ERROR) and also
occurs quite frequently when there is a lot of range merging activity.

Release note: None

a-robinson reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, 2 of 2 files at r3, 1 of 1 files at r4.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/29000#-:-LKbTTo_0jmSUwhESnEI:bg7igkv)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)

___
*[pkg/storage/replica_command.go, line 1050 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/29000#-LKbRvLx3tzlLjVugyod:-LKbRvLx3tzlLjVugyoe:b-vesh5q) ([raw file](https://github.com/cockroachdb/cockroach/blob/e39f6bd012b53c41b555a5d5dab0e436f357d9f6/pkg/storage/replica_command.go#L1050)):*
> ```Go
> 	}
> 
> 	every := log.Every(time.Minute)
> ```

This loop (and the `for len(removeTargets) > 0` loop below) really needs a reasonable backoff on retries. I'll send that out separately, though.


<!-- Sent from Reviewable.io -->

tschottdorf commented:
TFTR!

@benesch mentioned that RelocateRange needed to be rewritten, so I didn't want to touch it too much. Definitely agree, though.

bors r=a-robinson

tschottdorf commented:
bors r-

didn't see CI was red.

tschottdorf commented:
bors r=a-robinson

benesch commented on pkg/storage/store.go:
```diff
@@ -2652,6 +2651,10 @@ func (s *Store) removeReplicaImpl(
 
+	// During merges, the context might have the subsuming range, so we explicitly
+	// log the replica to be removed.
+	log.Infof(ctx, "removing replica r%d/%d", rep.RangeID, replicaID)
```
Thanks for this! I've been meaning to clean this one up for a while now.

a-robinson commented:
I assume you want to backport this? I want to backport #29013 and it's easiest to do them together (or at least to do this first).
